conda/plugins/hookspec.py

```diff
@@ -173,24 +173,27 @@ def conda_post_commands():
         """
 
     @_hookspec
-    def conda_auth(self) -> Iterable[CondaAuth]:
+    def conda_auth_handlers(self) -> Iterable[CondaAuthHandler]:
         """
-        Register conda auth hook. This plugin hook replaces the allows authors
-        to add new ``request.AuthBase`` subclasses which can then be used by individual
-        channels while authenticated to HTTP/HTTPS services.
+        Register a conda auth handler derived from the requests API.
+
+        This plugin hook allows attaching requests auth handler subclasses,
+        e.g. when authenticating requests against individual channels hosted
+        at HTTP/HTTPS services.
 
         **Example:**
 
         .. code-block:: python
 
+            import os
             from conda import plugins
             from requests.auth import AuthBase
 
 
-            class CustomAuth(AuthBase):
+            class EnvironmentHeaderAuth(AuthBase):
```

Comment:
I updated the example here since I think we'd set a bad example if we'd hard-code username/password in the example, and it will be rendered in the docs after all. Environment variable based auth is not a lot worse, but at least it's an accepted evil following the 12 factor principles.